socos/music_lib.py

```diff
@@ -1,350 +1,93 @@
-                # dict as json. See self.index for details on fields.
-                values = [getattr(item, field) for field in
-                          fields[:-1]]
-                values.append(json.dumps(item.to_dict))
-                self.cursor.execute(query, values)
-            self.connection.commit()
-
-            # Print out status while running because indexing tracks can take a
-            # while
-            count += search['number_returned']
-            yield '{{: >3}}%  {{: >{0}}} out of {{: >{0}}}'\
-                .format(len(str(total)))\
-                .format(count * 100 / total, count, total)
-
-    def _get_columns(self, table):
-        """Return the names of the columns in the table"""
-        query = 'PRAGMA table_info({})'.format(table)
-        self.cursor.execute(query)
-        # The table descriptions look like: (0, u'title', u'text', 0, None, 0)
-        return [element[1] for element in self.cursor.fetchall()]
+    """ Class that implements music library support for socos """
 
     def tracks(self, sonos, *args):
-        """Search for and possibly play tracks from the music library
-
-        Usage: ml_tracks [field=]text [action] [number]
-
-        Field can be 'title', 'album' or 'artist'. If field is not given, then
-        'title' is used. Only a single word can be used as search text. Action
-        can be 'add' or 'replace' and number refers to the item number in the
-        search results.
-
-        Examples:
-        ml_tracks artist=metallica
-        ml_tracks unforgiven
-        ml_tracks unforgiven add 4
-        """
-        for string in self._search_and_play(sonos, 'tracks', *args):
-            yield string
-
-    def albums(self, sonos, *args):
-        """Search for and possibly play albums from the music library
-
-        Usage: ml_albums [field=]text [action] [number]
-
-        Field can be 'title' or 'artist'. If field is not given, then 'title'
-        is used. Only a single word can be used as search text. Action can be
-        'add' or 'replace' and number refers to the item number in the search
-        results.
-
-        Examples:
-        ml_albums artist=metallica
-        ml_albums black
-        ml_albums black add 1
-        """
-        for string in self._search_and_play(sonos, 'albums', *args):
-            yield string
-
-    def artists(self, sonos, *args):
-        """Search for and possibly play all by artists from music library
-
-        Usage: ml_artists text [action] [number]
-
-        'text' is searched for in the artist titles. Only a single word can '\
-        'be used as search text. Action can be 'add' or 'replace' and number '\
-        'refers to the item number in the search results.
-
-        Examples:
-        ml_artists metallica
-        ml_artists metallica add 1
-        """
-        for string in self._search_and_play(sonos, 'artists', *args):
-            yield string
+        """ Get available tracks of a device / groups """
```

Comment:
With this one, as well as all the rest of the "commands" I could use an explanation of the arguments especially what may be contained in `*args`. I think it could be done with a good doc string on _search_and_play and then refer to that from these specific methods